src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -3840,10 +3840,10 @@ encode %{
     __ tbnz(disp_hdr, exact_log2(markWord::monitor_value), object_has_monitor);
 
     if (!UseHeavyMonitors) {
-      if (LockingMode == 2) {
+      if (LockingMode == LIGHTWEIGHT) {
```

Comment:
You should consider changing uses of `UseHeavyMonitors` to
the appropriate check of `LockingMode`. For this case, use:
`    if (LockingMode != MONITOR) {`

You'll also need to change the implementation of the `UseHeavyMonitors`
option to set `LockingMode = MONITOR` or something like that.